### PR TITLE
[plugin.video.nhlgcl] 2017.10.9

### PR DESCRIPTION
**Pull request by eracknaphobia**
### Description

- Added Vegas Golden Knights
- Updated stream url's
### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: plugin.video.nhlgcl/resources/lib/nhl_tv.py [hunk 1]
```diff
@@ -473,115 +485,115 @@ def fetchStream(game_id, content_id,event_id):
...
         #Org
-        url = 'https://mf.svc.nhl.com/ws/media/mf/v2.4/stream?contentId='+content_id+'&playbackScenario=HTTP_CLOUD_TABLET_60&platform='+PLATFORM+'&sessionKey='+urllib.quote_plus(session_key)        
-        req = urllib2.Request(url)       
+        url = 'https://mf.svc.nhl.com/ws/media/mf/v2.4/stream?contentId='+content_id+'&playbackScenario='+PLAYBACK_SCENARIO+'&platform='+PLATFORM+'&sessionKey='+urllib.quote_plus(session_key)
+        req = urllib2.Request(url)
         req.add_header("Accept", "*/*")
         req.add_header("Accept-Encoding", "deflate")
-        req.add_header("Accept-Language", "en-US,en;q=0.8")                       
+        req.add_header("Accept-Language", "en-US,en;q=0.8")
         req.add_header("Connection", "keep-alive")
```
Comment (romanvm) [inline plugin.video.nhlgcl/resources/lib/nhl_tv.py]: I seriously doubt that vanilla Python `urllib2` support persistent connections.

File: plugin.video.nhlgcl/resources/lib/nhl_tv.py [hunk 2]
```diff
@@ -473,115 +485,115 @@ def fetchStream(game_id, content_id,event_id):
     i=0
     for i in range (0,9):
         #Org
-        url = 'https://mf.svc.nhl.com/ws/media/mf/v2.4/stream?contentId='+content_id+'&playbackScenario=HTTP_CLOUD_TABLET_60&platform='+PLATFORM+'&sessionKey='+urllib.quote_plus(session_key)        
-        req = urllib2.Request(url)       
+        url = 'https://mf.svc.nhl.com/ws/media/mf/v2.4/stream?contentId='+content_id+'&playbackScenario='+PLAYBACK_SCENARIO+'&platform='+PLATFORM+'&sessionKey='+urllib.quote_plus(session_key)
+        req = urllib2.Request(url)
         req.add_header("Accept", "*/*")
         req.add_header("Accept-Encoding", "deflate")
```
Comment (romanvm) [inline plugin.video.nhlgcl/resources/lib/nhl_tv.py]: I don't see that you decompress deflate-encoded response anywhere in your addon.

File: plugin.video.nhlgcl/resources/lib/nhl_tv.py [hunk 3]
```diff
@@ -590,83 +602,83 @@ def login():
 
         #Get Token
         url = 'https://user.svc.nhl.com/oauth/token?grant_type=client_credentials'
-        req = urllib2.Request(url)       
+        req = urllib2.Request(url)
         req.add_header("Accept", "application/json")
         req.add_header("Accept-Encoding", "gzip, deflate, sdch")
```
Comment (romanvm) [inline plugin.video.nhlgcl/resources/lib/nhl_tv.py]: Again, I don't see that you unpack compressed responses anywhere.

**Comment by eracknaphobia** [comment]
I've fixed the codeacy bot issue, somehow an old unused file got back in. As for the other issues, while I agree they should be fixed, they're not show stoppers and right now the addon doesn't work unless the other changes go live. I'm in the process of changing those urllib2 to requests, but it's going to take time and like I said the addon is broken at the moment.

**Comment by romanvm** [comment]
As for headers, it's more a recommendation, your server obviously do not respect `Accept-Encoding` client headers, otherwise you would have problems.
The addon looks OK except for multiple commits. Please squash them into one.

File: plugin.video.nhlgcl/resources/lib/globals.py [hunk 4]
```diff
@@ -8,7 +8,7 @@
 import json
 import cookielib
 import time
-from bs4 import BeautifulSoup 
+from bs4 import BeautifulSoup
```
Comment (Lunatixz) [inline plugin.video.nhlgcl/resources/lib/globals.py]: Unused import?

File: plugin.video.nhlgcl/resources/lib/globals.py [hunk 5]
```diff
@@ -177,12 +180,12 @@ def get_params():
 def addStream(name,link_url,title,game_id,epg,icon=None,fanart=None,info=None,video_info=None,audio_info=None,teams_stream=None,stream_date=None):
     ok=True
     u=sys.argv[0]+"?url="+urllib.quote_plus(link_url)+"&mode="+str(104)+"&name="+urllib.quote_plus(name)+"&game_id="+urllib.quote_plus(str(game_id))+"&epg="+urllib.quote_plus(str(epg))+"&teams_stream="+urllib.quote_plus(str(teams_stream))+"&stream_date="+urllib.quote_plus(str(stream_date))
-    
+
     if icon != None:
-        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=icon) 
+        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=icon)
```
Comment (Lunatixz) [inline plugin.video.nhlgcl/resources/lib/globals.py]: Can we move this to setart? IconImage and thumbnailImage were deprecated jarvis and up :)
http://mirrors.xbmc.org/docs/python-docs/16.x-jarvis/xbmcgui.html#ListItem

File: plugin.video.nhlgcl/resources/lib/globals.py [hunk 6]
```diff
@@ -177,12 +180,12 @@ def get_params():
...
     else:
-        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=ICON) 
-    
+        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=ICON)
```
Comment (Lunatixz) [inline plugin.video.nhlgcl/resources/lib/globals.py]: ''

File: plugin.video.nhlgcl/resources/lib/globals.py [hunk 7]
```diff
@@ -198,21 +201,21 @@ def addStream(name,link_url,title,game_id,epg,icon=None,fanart=None,info=None,vi
...
 
 def addFavToday(name,title,icon,fanart=None):
     info = {'plot':'','tvshowtitle':'NHL','title':name,'originaltitle':name,'aired':'','genre':'Sports'}
     audio_info, video_info = getAudioVideoInfo()
-    ok=True    
+    ok=True
     url = sys.argv[0] + '?url=/favteamCurrent&mode=510'
     if icon != None:
-        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=icon) 
+        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=icon)
```
Comment (Lunatixz) [inline plugin.video.nhlgcl/resources/lib/globals.py]: ''

File: plugin.video.nhlgcl/resources/lib/globals.py [hunk 8]
```diff
@@ -198,21 +201,21 @@ def addStream(name,link_url,title,game_id,epg,icon=None,fanart=None,info=None,vi
...
     else:
-        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=ICON) 
-    
+        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=ICON)
```
Comment (Lunatixz) [inline plugin.video.nhlgcl/resources/lib/globals.py]: ''

File: plugin.video.nhlgcl/resources/lib/globals.py [hunk 9]
```diff
@@ -229,20 +232,20 @@ def addFavToday(name,title,icon,fanart=None):
...
     xbmcplugin.setContent(addon_handle, 'tvshows')
-    
+
     return ok
 
 
 def addLink(name,url,title,iconimage,info=None,video_info=None,audio_info=None,fanart=None):
     ok=True
-    liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=iconimage)    
+    liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=iconimage)
```
Comment (Lunatixz) [inline plugin.video.nhlgcl/resources/lib/globals.py]: ''

File: plugin.video.nhlgcl/resources/lib/globals.py [hunk 10]
```diff
@@ -229,20 +232,20 @@ def addFavToday(name,title,icon,fanart=None):
...
     liz.setProperty("IsPlayable", "true")
     liz.setInfo( type="Video", infoLabels={ "Title": title } )
     liz.setProperty('fanart_image', FANART)
     if iconimage != None:
-        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=iconimage) 
+        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=iconimage)
```
Comment (Lunatixz) [inline plugin.video.nhlgcl/resources/lib/globals.py]: ''

File: plugin.video.nhlgcl/resources/lib/globals.py [hunk 11]
```diff
@@ -229,20 +232,20 @@ def addFavToday(name,title,icon,fanart=None):
...
     else:
-        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=ICON) 
+        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=ICON)
```
Comment (Lunatixz) [inline plugin.video.nhlgcl/resources/lib/globals.py]: ''

File: plugin.video.nhlgcl/resources/lib/globals.py [hunk 12]
```diff
@@ -276,9 +279,9 @@ def addDir(name,url,mode,iconimage,fanart=None,game_day=None):
         u = u+"&game_day="+urllib.quote_plus(game_day)
 
     if iconimage != None:
-        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=iconimage) 
+        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=iconimage)
```
Comment (Lunatixz) [inline plugin.video.nhlgcl/resources/lib/globals.py]: ''

File: plugin.video.nhlgcl/resources/lib/globals.py [hunk 13]
```diff
@@ -276,9 +279,9 @@ def addDir(name,url,mode,iconimage,fanart=None,game_day=None):
...
     else:
-        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=ICON) 
+        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=ICON)
```
Comment (Lunatixz) [inline plugin.video.nhlgcl/resources/lib/globals.py]: ''

File: plugin.video.nhlgcl/resources/lib/globals.py [hunk 14]
```diff
@@ -288,20 +291,20 @@ def addDir(name,url,mode,iconimage,fanart=None,game_day=None):
...
 
 
-def addPlaylist(name,game_day,url,mode,iconimage,fanart=None):       
+def addPlaylist(name,game_day,url,mode,iconimage,fanart=None):
     ok=True
     u=sys.argv[0]+"?url="+urllib.quote_plus(url)+"&mode="+str(mode)+"&name="+urllib.quote_plus(name)+"&icon="+urllib.quote_plus(iconimage)
 
     if iconimage != None:
-        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=iconimage) 
+        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=iconimage)
```
Comment (Lunatixz) [inline plugin.video.nhlgcl/resources/lib/globals.py]: ''

File: plugin.video.nhlgcl/resources/lib/globals.py [hunk 15]
```diff
@@ -288,20 +291,20 @@ def addDir(name,url,mode,iconimage,fanart=None,game_day=None):
...
     else:
-        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=ICON) 
+        liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=ICON)
```
Comment (Lunatixz) [inline plugin.video.nhlgcl/resources/lib/globals.py]: ''

File: plugin.video.nhlgcl/addon.xml [hunk 16]
```diff
@@ -1,26 +1,23 @@
...
-<addon id="plugin.video.nhlgcl" name="NHL TV™" version="2017.2.24" provider-name="eracknaphobia">
+<addon id="plugin.video.nhlgcl" name="NHL TV™" version="2017.10.3" provider-name="eracknaphobia">
   <requires>
-   <import addon="xbmc.python" version="2.24.0"/>   
-   <import addon="script.module.simplejson" version="3.3.0"/>  
-   <import addon="script.module.pytz" version="2014.2" />   
-   <import addon="script.module.beautifulsoup4" version="4.3.1" /> 
-   <import addon="script.module.pil" version="1.1.7" /> 
+   <import addon="xbmc.python" version="2.24.0"/>
+   <import addon="script.module.simplejson" version="3.3.0"/>
```
Comment (Lunatixz) [inline plugin.video.nhlgcl/addon.xml]: Json should have been included with jarvis, don't think this module is necessary. :)

File: plugin.video.nhlgcl/resources/lib/globals.py (hunk 4 above)
Comment (eracknaphobia) [reply plugin.video.nhlgcl/resources/lib/globals.py]: This is used in nhl_tv.py

File: plugin.video.nhlgcl/resources/lib/globals.py (hunk 5 above)
Comment (eracknaphobia) [reply plugin.video.nhlgcl/resources/lib/globals.py]: Will do.

File: plugin.video.nhlgcl/addon.xml (hunk 16 above)
Comment (eracknaphobia) [reply plugin.video.nhlgcl/addon.xml]: I'll remove unnecessary module import

**Comment by eracknaphobia** [comment]
@romanvm For the Accept-Encoding header is it better to switch these all to identity or just omit them when using python requests module? 

@romanvm @Lunatixz I made a bunch more changes and I will try to update / squash this pull request, but being the git noob that I am, if I can't get it to work I may just close and open a new pull request.

**Comment by learningit** [comment]
@eracknaphobia just my 2 cents here. In my experience it often doesn't matter how you set Accept_Encoding header, particularly if you are trying to disable gzip encoding. Either the website or a router in between may just compress. So it's always good to check Content-Encoding for compression.

It's pretty straightforward and you don't have to rewrite for the requests module. An example is here:
[https://github.com/xbmc/repo-scripts/blob/33b643cc1649ea017461948f0f755c2e3f95484e/script.module.t1mlib/lib/t1mlib.py#L67](url)

**Comment by learningit** [comment]
Also here's a good explanation of doing a squash:
[https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit](url)

**Comment by eracknaphobia** [comment]
@learningit Thanks! Looks like requests does that automatically, so I should be ok. 
http://docs.python-requests.org/en/master/community/faq/#encoded-data

Thanks for the link to the squash doc, I'll give it a try.

**Pull request closed without merging**